src/storage/transaction/TransactionManager.cpp

```diff
@@ -211,54 +218,69 @@ void TransactionManager::onLeaderElectedWrapper(
+  TermID prevTerm = itCurrTerm == currTerm_.end() ? -1 : itCurrTerm->second;
+
   std::unique_ptr<kvstore::KVIterator> iter;
-  auto prefix = ConsistUtil::primePrefix(partId);
-  auto rc = env_->kvstore_->prefix(spaceId, partId, prefix, &iter);
-  if (rc == nebula::cpp2::ErrorCode::SUCCEEDED) {
-    for (; iter->valid(); iter->next()) {
-      auto edgeKey = ConsistUtil::edgeKeyFromPrime(iter->key()).str();
-      VLOG(1) << "scanned prime edge: " << folly::hexlify(edgeKey);
-      auto lk = getLockCore(spaceId, partId, termId, false);
-      auto succeed = lk->try_lock(edgeKey);
-      if (!succeed) {
-        LOG(ERROR) << "not supposed to lock fail: "
-                   << ", spaceId " << spaceId << ", partId " << partId << ", termId " << termId
-                   << folly::hexlify(edgeKey);
-      }
-      addPrime(spaceId, partId, termId, edgeKey, ResumeType::RESUME_CHAIN);
-    }
-  } else {
-    VLOG(1) << "primePrefix() " << apache::thrift::util::enumNameSafe(rc);
-  }
+  for (auto i = 0U; i != prefixVec.size(); ++i) {
+    auto rc = env_->kvstore_->prefix(spaceId, partId, prefixVec[i], &iter);
+    if (rc == nebula::cpp2::ErrorCode::SUCCEEDED) {
+      for (; iter->valid(); iter->next()) {
+        auto edgeKey = ConsistUtil::edgeKeyFromPrime(iter->key()).str();
+        VLOG(1) << "scanned prime edge: " << folly::hexlify(edgeKey);
+        auto lk = getLockCore(spaceId, partId, termId, false);
+        auto succeed = lk->try_lock(edgeKey);
+        if (!succeed) {
+          LOG(ERROR) << "not supposed to lock fail: "
+                     << ", spaceId " << spaceId << ", partId " << partId << ", termId " << termId
+                     << folly::hexlify(edgeKey);
+        }
+        auto prevMemLockKey = std::make_tuple(spaceId, partId, prevTerm);
+        auto prevMemLockIter = memLocks_.find(prevMemLockKey);
+        auto spLock = prevMemLockIter == memLocks_.end() ? nullptr : prevMemLockIter->second;
+        auto hasUnfinishedTask = [lk = spLock, key = edgeKey] {
+          return (lk != nullptr) && lk->contains(key);
+        };
 
-  prefix = ConsistUtil::doublePrimePrefix(partId);
-  rc = env_->kvstore_->prefix(spaceId, partId, prefix, &iter);
-  if (rc == nebula::cpp2::ErrorCode::SUCCEEDED) {
-    for (; iter->valid(); iter->next()) {
-      auto edgeKey = ConsistUtil::edgeKeyFromDoublePrime(iter->key()).str();
-      VLOG(1) << "scanned double prime edge: " << folly::hexlify(edgeKey);
-      auto lk = getLockCore(spaceId, partId, termId, false);
-      auto succeed = lk->try_lock(edgeKey);
-      if (!succeed) {
-        LOG(ERROR) << "not supposed to lock fail: "
-                   << ", space " << spaceId << ", partId " << partId << ", termId " << termId
-                   << folly::hexlify(edgeKey);
+        if (!hasUnfinishedTask()) {
+          addPrime(spaceId, partId, termId, edgeKey, resumeVec[i]);
+        } else {
+          folly::Promise<folly::Unit> pro;
+          auto fut = pro.getFuture();
```

Comment:
minor: "pro" usually means good vs. con. It's better to use "p", "promise" or "xxPromise" to indicate promise.